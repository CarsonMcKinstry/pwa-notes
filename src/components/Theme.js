import React, {Component} from 'react';
import {PropTypes} from 'prop-types';
import { ThemeProvider, injectGlobal } from 'styled-components';

const theme = {
  primary: {
    base: '#1e88e5',
    light: '#6ab7ff',
    dark: '#005cb2'
  },
  danger: {
    base: '#f44336',
    light: '#ff7961',
    dark: '#ba000d'
  }
}

injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Roboto|Material+Icons');

  html, body {
    margin: 0;
    padding: 0;
    font-family: 'Roboto', sans-serif;
  }
  
  :root {
    --mdc-theme-primary: ${theme.primary.base}!important;
  }

`


const Theme = ({children, ...props}) => (
  <ThemeProvider theme={{
    ...props.theme,
    ...theme
  }}>
    {children}
  </ThemeProvider>
)

Theme.propTypes = {}

export default Theme;