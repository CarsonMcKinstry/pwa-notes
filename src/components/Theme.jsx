/* eslint-disable no-unused-expressions */
import React from 'react';
import { ThemeProvider, injectGlobal } from 'styled-components';

export const baseTheme = {
  primary: {
    base: '#0d47a1',
    light: '#5472d3',
    dark: '#002171'
  },
  secondary: {
    base: '#81c784',
    light: '#b2fab4',
    dark: '#519657'
  },
  danger: {
    base: '#f44336',
    light: '#ff7961',
    dark: '#ba000d'
  }
};

injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Roboto|Material+Icons');

  html, body {
    margin: 0;
    padding: 0;
    font-family: 'Roboto', sans-serif;
    height: 100%;
  }

  #root {
    height: 100%;
  }
  
  :root {
    --mdc-theme-primary: ${baseTheme.primary.base}!important;
    --mdc-theme-secondary: ${baseTheme.secondary.base}!important;
  }

`;


const Theme = ({ children, theme }) => (
  <ThemeProvider
    theme={{
      ...theme,
      ...baseTheme
    }}
  >
    {children}
  </ThemeProvider>
);

export default Theme;
