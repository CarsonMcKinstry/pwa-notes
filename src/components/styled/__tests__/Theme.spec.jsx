/* global it, describe, expect, shallow, mount */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import Theme, { baseTheme } from '../Theme';

describe('<Theme/>', () => {
  it('should render without crashing', () => {
    const tree = shallow(<Theme />);

    expect(tree).toMatchSnapshot();
  });

  it('should render children', () => {
    const tree = shallow(
      <Theme>
        <div />
      </Theme>
    );
    expect(tree.find('div').length).toEqual(1);
    expect(tree).toMatchSnapshot();
  });

  it('should have a ThemeProvder with a theme of the baseTheme', () => {
    const tree = mount(
      <Theme />
    );

    expect(tree.find(ThemeProvider).length).toEqual(1);
    expect(tree.find(ThemeProvider).props().theme).toEqual(baseTheme);
    expect(tree).toMatchSnapshot();
  });

  it('should extend the baseTheme', () => {
    const theme = {
      white: '#fff',
      black: '#000'
    };
    const tree = mount(
      <Theme theme={theme} />
    );

    expect(tree.find(ThemeProvider).props().theme).toEqual({
      ...baseTheme,
      ...theme
    });
    expect(tree).toMatchSnapshot();
  });
});
