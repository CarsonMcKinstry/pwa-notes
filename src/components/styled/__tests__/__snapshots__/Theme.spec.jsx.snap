// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Theme/> should extend the baseTheme 1`] = `
<Theme
  theme={
    Object {
      "black": "#000",
      "white": "#fff",
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "black": "#000",
        "danger": Object {
          "base": "#f44336",
          "dark": "#ba000d",
          "light": "#ff7961",
        },
        "primary": Object {
          "base": "#0d47a1",
          "dark": "#002171",
          "light": "#5472d3",
        },
        "secondary": Object {
          "base": "#81c784",
          "dark": "#519657",
          "light": "#b2fab4",
        },
        "white": "#fff",
      }
    }
  />
</Theme>
`;

exports[`<Theme/> should have a ThemeProvder with a theme of the baseTheme 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "danger": Object {
          "base": "#f44336",
          "dark": "#ba000d",
          "light": "#ff7961",
        },
        "primary": Object {
          "base": "#0d47a1",
          "dark": "#002171",
          "light": "#5472d3",
        },
        "secondary": Object {
          "base": "#81c784",
          "dark": "#519657",
          "light": "#b2fab4",
        },
      }
    }
  />
</Theme>
`;

exports[`<Theme/> should render children 1`] = `
<ThemeProvider
  theme={
    Object {
      "danger": Object {
        "base": "#f44336",
        "dark": "#ba000d",
        "light": "#ff7961",
      },
      "primary": Object {
        "base": "#0d47a1",
        "dark": "#002171",
        "light": "#5472d3",
      },
      "secondary": Object {
        "base": "#81c784",
        "dark": "#519657",
        "light": "#b2fab4",
      },
    }
  }
>
  <div />
</ThemeProvider>
`;

exports[`<Theme/> should render without crashing 1`] = `
<ThemeProvider
  theme={
    Object {
      "danger": Object {
        "base": "#f44336",
        "dark": "#ba000d",
        "light": "#ff7961",
      },
      "primary": Object {
        "base": "#0d47a1",
        "dark": "#002171",
        "light": "#5472d3",
      },
      "secondary": Object {
        "base": "#81c784",
        "dark": "#519657",
        "light": "#b2fab4",
      },
    }
  }
/>
`;
